### DISCUSSION TITLE
POD performance benchmark for large matrix

### DISCUSSION THREAD

**Discussion by jacktang** [pymor/pymor#1991] [Q&A] [answered]
Hello there,

Any POD performance benchmark for large matrix? say the shape is `10000x10000`.
Also I noticed there are some matrix equation solver like slycot and mess, how I config it to enable them if I want to get the performance? Thanks!

**Comment by pmli** [comment, marked as answer]
> Any POD performance benchmark for large matrix? say the shape is `10000x10000`.

There are no POD performance benchmarks in pyMOR, but I would say that [HAPOD](https://docs.pymor.org/2022-2-0/autoapi/pymor/algorithms/hapod/index.html) and [randomized methods](https://docs.pymor.org/2022-2-0/autoapi/pymor/algorithms/rand_la/index.html) are appropriate for a matrix of such size.

> Also I noticed there are some matrix equation solver like slycot and mess, how I config it to enable them if I want to get the performance? Thanks!

By default, pyMOR will use `pymess` if it is installed, otherwise it will use `slycot` if installed. The solver can also be selected using via defaults, e.g.

```python
from pymor.core.defaults import set_defaults
set_defaults({'pymor.algorithms.lyapunov.solve_cont_lyap_dense.default_solver_backend': 'slycot'})
```

Note that the dense matrix equation solvers do not necessarily give better performance compared to the default solver from SciPy, but they should be more accurate. On the other hand, low-rank solvers in pyMOR and `pymess` do provide performance improvements for large enough problems.

**Reply by jacktang** [reply]
@pmli Thanks! It helps a lot.
